src/Archives/ClmFile.cpp

```diff
@@ -112,7 +112,11 @@ namespace Archives
 	{
 		CheckPackedFileIndexBounds(fileIndex);
 
-		throw std::logic_error("OpenStream not yet implemented for Clm files.");
+		FileSliceReader reader = clmFileReader.Slice(
+			indexEntries[fileIndex].dataOffset,
+			indexEntries[fileIndex].dataLength);
+
+		return std::make_unique<FileSliceReader>(reader);
```

Comment:
I'm wondering if there are performance consequences of this, and if the result of `Slice` should be stored immediately into such a variable, or even constructed as part of the return statement. I'm guessing there are a few constructor and copy constructor calls.

Reply:
I didn't create the slice inside the make_unique call because it was making the line of code long and more difficult to understand. So readability drove the separation. If there are performance concerns, we can combine it, as it wouldn't be too hard to read.